### PR TITLE
New webview provider

### PR DESCRIPTION
**Pull request by CiprianDraghici**
### Reproduce
Issue exists on version `2.` of sdk-dapp.

### Contains breaking changes
[x] No

[] Yes

### Updated CHANGELOG
[x] Yes

### Testing
[x] User testing
[] Unit tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,8 +26,7 @@
     "storybook": "start-storybook -p 6006",
     "test-storybook": "test-storybook",
     "build-storybook": "build-storybook",
-    "lint": "eslint --fix --ext js,ts,tsx src",
-    "pre-release": "node prerelease-script.js"
```
Comment (arhtudormorar) [inline package.json:30]: why remove?

File: src/providers/webviewProvider/NewWebviewProvider.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  SignMessageStatusEnum
+} from '@multiversx/sdk-web-wallet-cross-window-provider/out/types';
+import { loginWithNativeAuthToken } from 'services/nativeAuth/helpers/loginWithNativeAuthToken';
+import { IDappProvider } from 'types/dappProvider.types';
+import { setExternalProviderAsAccountProvider } from '../accountProvider';
+import { getTargetOrigin } from './targetOrigin';
+
+type SendPostMessageType<T extends CrossWindowProviderRequestEnums> = {
+  type: T;
+  payload?: any;
```
Comment (arhtudormorar) [inline src/providers/webviewProvider/NewWebviewProvider.ts:17]: not any

File: src/providers/webviewProvider/NewWebviewProvider.ts [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+    const { data, error } = response.payload;
+
+    if (error || !data) {
+      console.error('Unable to sign transactions');
+      return null;
+    }
+
+    const transactions = data;
+    return transactions.map((tx: any) => Transaction.fromPlainObject(tx));
```
Comment (arhtudormorar) [inline src/providers/webviewProvider/NewWebviewProvider.ts:88]: not any

File: src/providers/webviewProvider/NewWebviewProvider.ts [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    type: T;
+    payload: ReplyWithPostMessageType<T>['payload'];
+  }> {
+    return await new Promise((resolve) => {
+      window.addEventListener(
+        'message',
+        async function eventHandler(
+          event: MessageEvent<{
+            type: T;
+            payload: any;
```
Comment (arhtudormorar) [inline src/providers/webviewProvider/NewWebviewProvider.ts:139]: check any

File: src/providers/webviewProvider/NewWebviewProvider.ts [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+          if (event.origin != getTargetOrigin()) {
+            console.error('origin not accepted', {
+              eventOrigin: event.origin
+            });
+            return;
+          }
+
+          const isCurrentAction =
+            action === type ||
+            type === CrossWindowProviderResponseEnums.cancelResponse; // TODO: check if this is needed
```
Comment (arhtudormorar) [inline src/providers/webviewProvider/NewWebviewProvider.ts:153]: remove comment

File: src/wrappers/DappProvider/DappProvider.tsx [hunk 6]
```diff
@@ -39,7 +40,8 @@ export const DappProvider = ({
   if (externalProvider != null) {
     setExternalProvider(externalProvider);
   } else if (dappConfig?.shouldUseWebViewProvider) {
-    setExternalProvider(webviewProvider);
+    // setExternalProvider(webviewProvider);
```
Comment (arhtudormorar) [inline src/wrappers/DappProvider/DappProvider.tsx:43]: Add TODO: ...

File: package.json (hunk 1 above)
Comment (CiprianDraghici) [reply package.json:30]: deprecated

File: src/wrappers/DappProvider/DappProvider.tsx [hunk 7]
```diff
@@ -39,7 +46,11 @@ export const DappProvider = ({
   if (externalProvider != null) {
```
Comment (arhtudormorar) [inline src/wrappers/DappProvider/DappProvider.tsx:46]: this logic here has if elseif. Can stay like this or have a small helper above that uses early returns

**Review by arhtudormorar — APPROVED** [review]

File: src/providers/webviewProvider/NewWebviewProvider.ts [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+      console.log({
+        error,
+        data
+      });
+      console.error('Unable to sign message');
+      return null;
+    }
+
+    if (data.status !== SignMessageStatusEnum.signed) {
+      console.log({
```
Comment (arhtudormorar) [inline src/providers/webviewProvider/NewWebviewProvider.ts:124]: log

File: src/providers/webviewProvider/NewWebviewProvider.ts [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+  async signMessage(message: SignableMessage): Promise<SignableMessage | null> {
+    const response = await this.sendPostMessage({
+      type: CrossWindowProviderRequestEnums.signMessageRequest,
+      payload: message
+    });
+
+    const { data, error } = response.payload;
+
+    if (error || !data) {
+      console.log({
```
Comment (arhtudormorar) [inline src/providers/webviewProvider/NewWebviewProvider.ts:115]: log

**Pull request merged**